src/lib/__tests__/RoutingManager-test.ts

```diff
@@ -443,6 +437,84 @@ describe('RoutingManager', () => {
         },
       });
     });
+
+    test('skips duplicate route state entries', async () => {
+      let triggerChange = false;
+      const searchClient = createSearchClient();
+      const stateMapping = createFakeStateMapping({
+        stateToRoute(uiState) {
+          if (triggerChange) {
+            return {
+              ...uiState,
+              indexName: {
+                ...uiState.indexName,
+                triggerChange,
+              },
+            };
+          }
+
+          return uiState;
+        },
+      });
+      const history = createFakeHistory();
+      const router = createFakeRouter({
+        onUpdate(fn) {
+          history.subscribe(state => {
+            fn(state);
+          });
+        },
+        write: jest.fn(state => {
+          history.push(state);
+        }),
+      });
+
+      const search = instantsearch({
+        indexName: 'indexName',
+        searchClient,
+        routing: {
+          router,
+          stateMapping,
+        },
+      });
+
+      const fakeSearchBox: any = createFakeSearchBox();
+      const fakeHitsPerPage = createFakeHitsPerPage();
+
+      search.addWidgets([fakeSearchBox, fakeHitsPerPage]);
+
+      search.start();
+
+      await runAllMicroTasks();
+
+      // Trigger an update - push a change
+      fakeSearchBox.refine('Apple');
+
+      expect(router.write).toHaveBeenCalledTimes(1);
+      expect(router.write).toHaveBeenLastCalledWith({
+        indexName: {
+          query: 'Apple',
+        },
+      });
+
+      // Trigger change without UI state change
+      search.removeWidgets([fakeHitsPerPage]);
+
+      expect(router.write).toHaveBeenCalledTimes(1);
+
+      await runAllMicroTasks();
+
+      triggerChange = true;
+      // Trigger change without UI state change but with a route change
+      search.removeWidgets([fakeHitsPerPage]);
```

Comment:
We removed `fakeHitsPerPage` before, and can we do it again here?

Reply:
It indeed makes more sense to remove another widget.

Addressed in a6b5dbd7efd8869a45ee884e4e1edf7fc51c732a.